File: Data Structures & Algorithms/string-encode-and-decode/submission-1.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
         string new_str;
        for(int i=0;i<strs.size();i++)
        {
            int length=strs[i].size();
           new_str+=to_string(length)+'#'+strs[i];
        }
        //cout<<new_str;
        return new_str;
    }

    vector<string> decode(string s) {
        int len=s.size();
        int i=0;
        vector<string> v;
        while(i<len)
        {
            string str="";
            while(s[i]!='#')
            {
                str+=s[i];
                i++;
            }
            int num=stoi(str);
            v.push_back(s.substr(i+1,num));
            i=i+1+num;
        }
        
        return v;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp=[-1]*(n+1)
        def func(i):
            if i>n:
                return 0
            if dp[i]!=-1:
                return dp[i]
            if i==n:
                return 1
            dp[i]=func(i+1)+func(i+2)
            return dp[i]
        return func(0)
        